### DISCUSSION TITLE
Adding new items to list - 'FieldUrlValue' is inaccessible due to its protection level

### DISCUSSION THREAD

**Discussion by thunderstorm654** [pnp/pnpcore#942] [Q&A] [answered]
I'm having trouble with the most basic things - I'm just trying to add new items to an existing list with existing fields, specifically an URL field.

With the following code as per example here https://pnp.github.io/pnpcore/using-the-sdk/listitems-fields.html#url-fields, I just get compiler errors:

```
// Create a list item key/value pair collection
Dictionary<string, object> values = new Dictionary<string, object>()
{
    { "Title", "Item1" },
};

// Option A: instantiate a FieldUrlValue instance
values.Add("MyField", new FieldUrlValue("https://aka.ms/m365pnp", "PnP Rocks!"));
```

I'm getting the compiler error `'FieldUrlValue' is inaccessible due to its protection level`.

What am I missing here, this is driving me crazy :)

Thanks!

EDIT: I found that this above sample code works if using the nightly version of the SDK. If I need to use the latest non-nightly version 1.7.0 for production release, how is it possible to add URL field values? Many thanks.

**Comment by jansenbe** [comment, marked as answer]
@thunderstorm654 : correct, we've simplified the model to allow direct instantiation of fieldvalue objects. See option B from https://pnp.github.io/pnpcore/using-the-sdk/listitems-fields.html#url-fields for the "old" method which still is fully supported. 

![image](https://user-images.githubusercontent.com/7451219/185889328-5d63aac8-56d0-42c7-ac19-5dc269eeae4f.png)

The old approach requires you to have a reference to the `IField` instance you're adding a value for, this used to be required in the past but with the recent refactoring we could simplify the model.
